### PR TITLE
Added preferred.js with tests and additional tests for OUListContent.js

### PR DESCRIPTION
**Pull request by varunsivashankar**
**preferred.js**
- This file contains a function called _createForm_, which accesses Chrome storage and displays the form with the names of the stored preferred OUs.
- It also contains the function _applyFunc_ that is called when the apply button is clicked. It gets the dataRowId from the name of the selected entity. 
- TODO: Message passing needs to be implemented such that the dataRowId found by _applyFunc_ gets sent to OUSelectContent.js.
- These functions were tested with the files **preferredTest.js** and **preferredTest.html**.

image

**OUListContentTest.js**
- Added tests for the function _monitorChanges_ which implements mutation observer.

image

### CODE REVIEW

File: STEP_Project/preferred.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		{
+			let elt = document.createElement('input');
+			elt.setAttribute("type", "radio");
+			elt.setAttribute("value", keys[i]);
+			elt.setAttribute("name", "preferred");
+
+			let label = document.createElement('label');
+			label.appendChild(elt);
+			let OUName = document.createTextNode(data[keys[i]]);
+	    	label.appendChild(OUName);
```
Comment (atishey-jain) [inline STEP_Project/preferred.js:29]: nit: Fix indentation

File: STEP_Project/preferred.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+			if (selectForm[i].checked == true)
+			{
+				dataRowId = selectForm[i].value;
+				break;
+			}
+		}
+
+		if (dataRowId == null)
+		{
+			alert('Please choose an OU');
```
Comment (atishey-jain) [inline STEP_Project/preferred.js:70]: I think we should disable the button until an orgUnit is selected through radio button.

Comment (varunsivashankar) [reply STEP_Project/preferred.js]: I fixed this by keeping the apply button disabled initially. I enable it when one of the options on the form has been selected (event listener).

File: STEP_Project/preferred.js (hunk 1 above)
Comment (varunsivashankar) [reply STEP_Project/preferred.js]: done

File: STEP_Project/preferred.js [hunk 3]
```diff
@@ -32,9 +32,32 @@ function createForm()
...
+        if (selectForm[i].checked == true)
+        {
+            let applyButton = document.getElementById("apply");
+            applyButton.disabled = false;
+            applyListener(applyButton); 
+            return;
+        }
+    }
+}
+
```
Comment (atishey-jain) [inline STEP_Project/preferred.js:60]: Add tests for this and update screenshot of tests in PR description.

Comment (varunsivashankar) [reply STEP_Project/preferred.js:60]: Added testing and uploaded the new screenshot

**Pull request merged**